src/main/java/org/opentripplanner/updater/bike_rental/GenericJsonBikeRentalDataSource.java

```diff
@@ -103,42 +103,41 @@ public boolean update() {
     }
 
     private void parseJSON(InputStream dataStream) throws IllegalArgumentException, IOException {
-
-        ArrayList<BikeRentalStation> out = new ArrayList<>();
-
         String rentalString = convertStreamToString(dataStream);
-
         ObjectMapper mapper = new ObjectMapper();
         JsonNode rootNode = mapper.readTree(rentalString);
-
         if (!jsonParsePath.equals("")) {
             String delimiter = "/";
             String[] parseElement = jsonParsePath.split(delimiter);
-            for(int i =0; i < parseElement.length ; i++) {
+            for(int i = 0; i < parseElement.length; i++) {
                 rootNode = rootNode.path(parseElement[i]);
             }
-
             if (rootNode.isMissingNode()) {
                 throw new IllegalArgumentException("Could not find jSON elements " + jsonParsePath);
-              }
+            }
         }
+        processNodes(rootNode);
+    }
 
-        for (int i = 0; i < rootNode.size(); i++) {
-            // TODO can we use foreach? for (JsonNode node : rootNode) ...
-            JsonNode node = rootNode.get(i);
-            if (node == null) {
-                continue;
-            }
+    /**
+     * This method can be overridden in the event that a subclass does not yield stations.
+     * @param rootNode should be an array of sub-objects
+     */
+    public void processNodes(JsonNode rootNode) {
+        List<BikeRentalStation> out = new ArrayList<>();
+        for (JsonNode node : rootNode) {
+            if (node == null) continue;
             BikeRentalStation brstation = makeStation(node);
-            if (brstation != null)
+            if (brstation != null) {
                 out.add(brstation);
+            }
         }
         synchronized(this) {
             stations = out;
         }
     }
 
-    private String convertStreamToString(java.io.InputStream is) {
+    public static String convertStreamToString(java.io.InputStream is) {
         java.util.Scanner scanner = null;
```

Comment:
`public`?

Reply:
I noticed that this was a sort of library method that was not tied to a particular instance of the class. So I added `static`, and `public` just thinking it could be reused elsewhere. But I guess it should be documented and moved over to `org.opentripplanner.util` if that's the case. Anyway I added some Javadoc on this since it's non-obvious what the Scanner is doing here.